txindex.go

```diff
@@ -57,27 +57,29 @@ func (lw *LibWallet) IndexTransactions(afterIndexing func()) error {
 		}
 	}
 
-	beginHeight, err := lw.txDB.ReadIndexingStartBlock()
+	beginHeight, err := wallet.txDB.ReadIndexingStartBlock()
 	if err != nil {
-		log.Errorf("Get tx indexing start point error: %v", err)
+		log.Errorf("[%d] Get tx indexing start point error: %v", wallet.ID, err)
 		return err
 	}
 
-	endHeight := lw.GetBestBlock()
+	endHeight := wallet.GetBestBlock()
 
-	startBlock := wallet.NewBlockIdentifierFromHeight(beginHeight)
-	endBlock := wallet.NewBlockIdentifierFromHeight(endHeight)
+	startBlock := w.NewBlockIdentifierFromHeight(beginHeight)
+	endBlock := w.NewBlockIdentifierFromHeight(endHeight)
 
 	defer func() {
-		afterIndexing()
-		count, err := lw.txDB.Count(txindex.TxFilterAll, &Transaction{})
+		waitGroup.Done()
```

Comment:
Use an if-else statement below to log the count after indexing, then move this `waitGroup.Done()` call to the end of this defer func.